### ISSUE TITLE
It supports multiple data source inputs and is used as a pure detection engine

### ISSUE THREAD

**Issue by sun-sun-sun** [corazawaf/coraza#916]
## Summary
I want to input into a DB (e.g. MySQL Clickhouse) to read the various parts of the HTTP that have been stored (http_url, http_reponse_body, http_param), in fact, offline detection logs, can we support it?

**Comment by jcchavezs** [comment]
Yeah this is a very valid use case, you just need to write a program that
reads from DB and calls coraza API to run the analysis. Have you find any
inconvenience doing so?
